### PR TITLE
fix: fix fullautocomplete

### PR DESCRIPTION
**Pull request by giuliano176**

### CODE REVIEW

**Review by dhavaldodiya — DISMISSED** [review]

**Review by beawar — CHANGES_REQUESTED** [review]
Considered the history of this FullAutocomplete, probably some additional tests are required.
Imho, when we fix a bug, moreover if directly in release, there should be at least a test for the bug. To be meticulous, there should be enough test to cover all the possible cases for the bug.
Finally, there is no issue ref to see what this fix is about.

**Comment by giuliano176** [comment]
> Considered the history of this FullAutocomplete, probably some additional tests are required. Imho, when we fix a bug, moreover if directly in release, there should be at least a test for the bug. To be meticulous, there should be enough test to cover all the possible cases for the bug. Finally, there is no issue ref to see what this fix is about.

- The issue of tests was discussed before this implementation, the common decision taken was that no tests would have been written as the component is problematic and needs to be rewritten, and there are no tests in the module in general. Since the choice widely backfired, we could maybe move this task to lower priority (and leaving groups broken until the next release) so that we will have the time to add all the tests required?

**Comment by beawar** [comment]
I can help you write tests, we did some in the new branches we are working on, and they all include the usage of the contact input component. We can take those tests as base point and adapt them to test the contact input only.

Can you describe the problem you fixed here, so that we can try to code it?

**Comment by giuliano176** [comment]
the issue being solved is the groups being broken because the 'full' attribute being passed as empty string instead of undefined. seen the history of this fix I would rather test also other cases, if you have a set of tests that can be adapted that would help a lot

**Comment by beawar** [comment]
We can take this one as example test. From this we can build the needed tests. Of course we will need to adapt it to use the contact input directly, but it should not be too hard 

https://github.com/zextras/carbonio-contacts-ui/blob/04bc0a5b73b6cb72b56a778b09eb377638ead3c1/src/v2/views/board/NewContactGroupBoard.test.tsx#L536-L563

File: src/integrations/tests/contact-input.test.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		setCurrentBoard: jest.fn(),
+		getBoardContext: jest.fn(),
+		getBoard: jest.fn(),
+		closeBoard: closeBoardFn ?? jest.fn()
+	});
+}
+
+beforeAll(() => {
+	spyUseBoardHooks();
+});
```
Comment (beawar) [inline src/integrations/tests/contact-input.test.tsx]: we left this useless code, it can be removed

**Review by beawar — APPROVED** [review]
As agreed with @giuliano176 , I consider these tests the base set to cover the fix of this PR. Additional tests will be implemented (and are required from now since there are some examples) each time the contact input will be modified.

**Review by dhavaldodiya — APPROVED** [review]

**Pull request merged**